readthedocs/search/api.py

```diff
@@ -35,14 +37,37 @@ def get_link(self, obj):
     def get_highlight(self, obj):
         highlight = getattr(obj.meta, 'highlight', None)
         if highlight:
-            if hasattr(highlight, 'content'):
-                # Change results to turn newlines in highlight into periods
-                # https://github.com/rtfd/readthedocs.org/issues/5168
-                highlight.content = [result.replace('\n', '. ') for result in highlight.content]
-            ret = highlight.to_dict()
-            log.debug('API Search highlight: %s', pformat(ret))
+            ret = utils._remove_newlines_from_dict(highlight.to_dict())
+            log.debug('API Search highlight [Page title]: %s', pformat(ret))
             return ret
 
+    def get_inner_hits(self, obj):
+        inner_hits = getattr(obj.meta, 'inner_hits', None)
+        if inner_hits:
+            sections = inner_hits.sections or []
+            domains = inner_hits.domains or []
+            all_results = itertools.chain(sections, domains)
+
+            sorted_results = [
```

Comment:
@stsewd 
Here, If I used a generator expression -- `test_search_works_with_title_query` and `test_search_works_with_sections_query` will fail.
I can't find the reason though. For now, I have changed them to list comprehension

Reply:
So, I wasn't able to run the test because there is an import error, my guess is that when the generator gets evaluated the object `inner_hits` has changed. You can confirm this if you do a copy of `inner_hits.sections` and `inner_hits.domains` before assign them.

Also, I'd just left the list comprehension, since we don't know when the generator gets evaluated by django rest.

Reply:
I will add comments there to avoid any confusion in the future.